monai/bundle/workflows.py

```diff
@@ -0,0 +1,268 @@
+                if logging_file == "configs/logging.conf":
+                    warnings.warn("default logging file in 'configs/logging.conf' not exists, skip logging.")
+                else:
+                    raise FileNotFoundError(f"can't find the logging config file: {logging_file}.")
+            else:
+                logger.info(f"set logging properties based on config: {logging_file}.")
+                fileConfig(logging_file, disable_existing_loggers=False)
+
+        self.parser = ConfigParser()
+        self.parser.read_config(f=config_file)
+        if meta_file is not None:
+            if not os.path.exists(meta_file):
+                if meta_file == "configs/metadata.json":
+                    warnings.warn("default metadata file in 'configs/metadata.json' not exists, skip loading.")
+                else:
+                    raise FileNotFoundError(f"can't find the metadata config file: {meta_file}.")
+            else:
+                self.parser.read_meta(f=meta_file)
+
+        # the rest key-values in the _args are to override config content
+        self.parser.update(pairs=override)
+        self.init_id = init_id
+        self.run_id = run_id
+        self.final_id = final_id
+        # set tracking configs for experiment management
+        if tracking is not None:
+            if isinstance(tracking, str) and tracking in DEFAULT_EXP_MGMT_SETTINGS:
+                settings_ = DEFAULT_EXP_MGMT_SETTINGS[tracking]
+            else:
+                settings_ = ConfigParser.load_config_files(tracking)
+            self.patch_bundle_tracking(parser=self.parser, settings=settings_)
+
+    def initialize(self) -> Any:
+        # reset the "reference_resolver" buffer at initialization stage
+        self.parser.parse(reset=True)
+        return self._run_expr(id=self.init_id)
+
+    def run(self) -> Any:
+        return self._run_expr(id=self.run_id)
+
+    def finalize(self) -> Any:
+        return self._run_expr(id=self.final_id)
+
+    def check_properties(self):
+        super().check_properties()
+        # also check whether the optional properties use correct ID name if existing
+        wrong_props = []
+        for n, p in self.properties.items():
+            if not p[BundleProperty.REQUIRED] and not self._check_optional_id(name=n, property=p):
+                wrong_props.append(n)
+        if wrong_props:
+            raise ValueError(f"loaded bundle defines the following optional properties with wrong ID: {wrong_props}")
+        # check validation `interval` property
+        if "train#handlers" in self.parser:
+            for h in self.parser["train#handlers"]:
+                if h["_target_"] == "ValidationHandler":
+                    interval = h.get("interval", None)
+                    if interval is not None and interval != "val_interval":
+                        raise ValueError("please use ID 'val_interval' to define validation interval in training.")
```

Comment:
```suggestion
                    if interval not in (None, "val_interval"):
                        raise ValueError("Validation interval in training must be defined with ID 'val_interval'.")
```

This is a rather deep property in a bundle that is being required here, could it be optional in some way? It should be mentioned in the docstring for this class as well.

Reply:
updated in the latest commit.
This `val_interval` was requested by MONAI FL module: https://github.com/Project-MONAI/model-zoo/issues/52#issuecomment-1209626220.
@holgerroth please feel free to correct me if it's not necessary anymore.
It's an optional requirement, means "only when user really defined the interval for validation, the ID must be 'val_interval'".

Thanks.